dom.go

```diff
@@ -1612,6 +1661,14 @@ type BasicElement struct {
 	*BasicNode
 }
 
+func (e *BasicElement) Attributes() []Attribute {
+	return attributeListToAttributes(e.Get("attributes"))
+}
+
+func (e *BasicElement) Dataset() []Attribute {
+	return dataListToAttributes(e.Get("attributes"))
```

Comment:
I see, makes sense, i will just cleanup the duplicated logic and use `dataset` instead. Its not necessary an important feature for me to be honest, but since I was adding the attributes piece, I figured I should do the same for dataset